src/Widgets/StressTesting.cpp

```diff
@@ -0,0 +1,487 @@
+    compiledCount++;
+    if (compiledCount == 3)
+    {
+        nextTest();
+    }
+}
+
+void StressTesting::stop()
+{
+    if (stopping)
+        return;
+    stopping = true;
+
+    delete generatorRunner;
+    delete userRunner;
+    delete stdRunner;
+    delete generatorCompiler;
+    delete userCompiler;
+    delete stdCompiler;
+
+    argumentsRange.clear();
+    currentValue.clear();
+
+    startButton->setDisabled(false);
+    stopButton->setDisabled(true);
+
+    generatorRunner = nullptr, userRunner = nullptr, stdRunner = nullptr, generatorCompiler = nullptr,
+    userCompiler = nullptr, stdCompiler = nullptr;
+
+    stopping = false;
+}
+
+void StressTesting::nextTest()
+{
+    for (int i = 0; i <= argumentsCount; i++)
+    {
+        if (i == argumentsCount && (argumentsCount != 0 || currentValue[0]))
+        {
+            stop();
+            log->info(tr("Stress Testing"), tr("All tests finished, no countertest found"));
+            return;
+        }
+        if (currentValue[i] != argumentsRange[i].second)
+        {
+            currentValue[i]++;
+            break;
+        }
+        currentValue[i] = argumentsRange[i].first;
+    }
+
+    QString arguments = pattern;
+
+    for (int i = 0; i < argumentsCount; i++)
+        arguments = arguments.arg(QString::number(currentValue[i]));
+
+    log->info(tr("Stress Testing"), tr("Running with arguments \"%1\"").arg(arguments));
+
+    runFinishedCount = 0;
+
+    generatorRunner = new Core::Runner(0);
+    connect(generatorRunner, &Core::Runner::runFinished, this, &StressTesting::onRunFinished);
+    connect(generatorRunner, &Core::Runner::runOutputLimitExceeded, this, &StressTesting::onRunOutputLimitExceeded);
+    connect(generatorRunner, &Core::Runner::runKilled, this, &StressTesting::onRunKilled);
+    generatorRunner->run(generatorTmpPath, "", "C++", "", arguments, "", SettingsHelper::getDefaultTimeLimit());
+}
+
+void StressTesting::onCompilationErrorOccurred(const QString &error)
+{
+    stop();
+    emit compilationErrorOccurred(error);
+}
+
+void StressTesting::onCompilationFailed(const QString &reason)
+{
+    stop();
+    emit compilationFailed(reason);
+}
+
+void StressTesting::onCompilationKilled()
+{
+    stop();
+    emit compilationKilled();
+}
+
+void StressTesting::onRunFinished(int index, const QString &out, const QString & /*unused*/, int exitCode,
+                                  qint64 timeUsed, bool tle)
+{
+    QString head;
+    if (index == 0)
+    {
+        head = tr("Generator");
+    }
+    else if (index == 1)
+    {
+        head = tr("User program");
+    }
+    else if (index == 2)
+    {
+        head = tr("Standard program");
+    }
+
+    if (exitCode == 0)
+    {
+        log->info(head, tr("Execution has finished in %1ms").arg(timeUsed));
+        if (index == 0)
+        { // Generator
+
+            in = out;
+
+            QString language = mainWindow->getLanguage();
+            stdRunner = new Core::Runner(1);
```

Comment:
You can modify or inherit from Runner a class `StressRunner` and move these logic inside it. That way you can also change enum types.

Reply:
Why do we need a new class?

Reply:
If you make SourceType as enum class it cannot be implicitly converted to int type. Runner takes an int argument hence to make it work you might have to inherit it. Since you used normal enum its fine.